File: 1201-1300/1260-Shift 2D Grid/1260-Shift 2D Grid.py
class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        nums = sum(grid, [])
        k %= len(nums)
        nums = nums[-k:] + nums[:-k]
        col = len(grid[0])
        return [nums[i:i + col] for i in range(0, len(nums), col)]

File: 1201-1300/1260-Shift 2D Grid/1260-Shift 2D Grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        int n = grid.size(), m = grid[0].size();
        vector<vector<int>> result(n, vector<int>(m));
        int start = m * n - k % (n * m);
        for (int i = start; i < start + n * m; ++i) {
            int row = i % (n * m) / m;
            int col = i % (n * m) % m;
            result[(i - start) / m][(i - start) % m] = grid[row][col];
        }
        
        return result;
    }
};
